sharding/export_test.go

```diff
@@ -0,0 +1,105 @@
+package sharding
+
+import "sync"
+
+func (nc *indexHashedNodesCoordinator) NodesConfig() map[uint32]*epochNodesConfig {
+	return nc.nodesConfig
+}
+
+func (ihgs *indexHashedNodesCoordinator) SaveState(key []byte) error {
+	return ihgs.saveState(key)
+}
+
+func (ihgs *indexHashedNodesCoordinator) CurrentEpoch() uint32 {
+	return ihgs.currentEpoch
+}
+
+func (ihgs *indexHashedNodesCoordinator) NodesCoordinatorToRegistry() *NodesCoordinatorRegistry {
+	return ihgs.nodesCoordinatorToRegistry()
+}
+
+func (ihgs *indexHashedNodesCoordinator) RegistryToNodesCoordinator(
+	config *NodesCoordinatorRegistry,
+) (map[uint32]*epochNodesConfig, error) {
+	return ihgs.registryToNodesCoordinator(config)
+}
+
+func (ihgs *indexHashedNodesCoordinator) ComputeShardForPublicKey(nodesConfig *epochNodesConfig) uint32 {
+	return ihgs.computeShardForPublicKey(nodesConfig)
+}
+
+func (ihgs *indexHashedNodesCoordinatorWithRater) ExpandEligibleList(validators []Validator, mut *sync.RWMutex) []Validator {
+	return ihgs.expandEligibleList(validators, mut)
+}
+
+func (enc *epochNodesConfig) ShardId() uint32 {
+	return enc.shardId
+}
+
+func (enc *epochNodesConfig) NbShards() uint32 {
+	return enc.nbShards
+}
+
+func (enc *epochNodesConfig) EligibleMap() map[uint32][]Validator {
+	return enc.eligibleMap
+}
+
+func (enc *epochNodesConfig) WaitingMap() map[uint32][]Validator {
+	return enc.waitingMap
+}
+
+func (enc *epochNodesConfig) MutNodesMaps() sync.RWMutex {
```

Comment:
This does not work as expected because it returns the mutex by value. Should have used pointers.
Might make the methods that output maps (that are protected by this mutex) to output copies of maps. In this way there will not be concurrency issues.

Reply:
Fixed by returning a pointer to the mutex.

Reply:
fixed by changing the package name and the export is no longer needed